influxdb3/src/commands/serve.rs

```diff
@@ -216,6 +217,67 @@ pub struct Config {
     /// for any hosts that share the same object store configuration, i.e., the same bucket.
     #[clap(long = "host-id", env = "INFLUXDB3_HOST_IDENTIFIER_PREFIX", action)]
     pub host_identifier_prefix: String,
+
+    /// The size of the in-memory Parquet cache in bytes.
+    #[clap(
+        long = "parquet-mem-cache-size",
+        env = "INFLUXDB3_PARQUET_MEM_CACHE_SIZE",
+        default_value_t = 1024 * 1024 * 1024,
+        action
+    )]
+    pub parquet_mem_cache_size: usize,
+
+    /// The percentage of entries to prune during a prune operation on the in-memory Parquet cache.
+    ///
+    /// This must be a number between 0 and 1.
+    #[clap(
+        long = "parquet-mem-cache-prune-percentage",
+        env = "INFLUXDB3_PARQUET_MEM_CACHE_PRUNE_PERCENTAGE",
+        default_value = "0.1",
+        action
+    )]
+    pub parquet_mem_cache_prune_percentage: PrunePercent,
+
+    /// The interval on which to check if the in-memory Parquet cache needs to be pruned.
+    #[clap(
+        long = "parquet-mem-cache-prune-interval",
+        env = "INFLUXDB3_PARQUET_MEM_CACHE_PRUNE_INTERVAL",
+        default_value = "10ms",
```

Comment:
Does this just grab a read lock? Seems like we don't need to check this so frequently by default, once a second seems more than enough?

Reply:
Yeah, only if it needs to prune would lock the inner map, but I agree, this is a little over-zealous.